### ISSUE TITLE
Text and word components are almost impossible to use in Firefox

### ISSUE THREAD

**Issue by louisremi** [byte-foundry/prototypo#211]
When typing text in the text/word preview components, the caret gets intermittently moved to the beginning of the text, which is rather frustrating.
This is due to the fact that the text is saved every 500ms (debounced). Once the text is saved, the textContent property of those elements is set to... exactly what it was. Chrome handles this "false change" without moving the caret position, but Firefox moves it to the beginning of the element (can't blame it).
@FranzPoize needs to look into the behavior of this components and make sure setting textContent is only used when appropriate.

Side note: the setupText method of these elements also defines the default text that they display. I'm pretty sure this default values belong to the model and shouldn't appear here. It'd be nice to have a single "default app values" object used only once in main.js

- [ ] fix the behavior of these components to make them usable in Firefox 
- [ ] move the default values somewhere else
- [ ] have a beer

**Comment by FranzPoize** [comment]
aebbe82bf02c84263e1ed49bb43b74f95d991ac6

Maybe it would be cool to separate the text component into a container and a text editable to allow for more control on the shouldComponentUpdate

**Comment by FranzPoize** [comment]
23364e4d4896494968c42e5e7280d5ca3096516f
